### PR TITLE
Add Filter by Epoch in kv/blocks.go 

### LINKED ISSUE #3949: Allow Filtering of Blocks and Block Roots by Epochs
This is pretty ugly:

```go
roots, err := db.BlockRoots(ctx, filters.NewFilter().SetStartSlot(helpers.StartSlot(checkpoint.Epoch)).SetEndSlot(helpers.StartSlot(checkpoint.Epoch+1)-1))
```
Instead, we should allow blocks to be filterable by `StartEpoch` and `EndEpoch` as well. Thanks @prestonvanloon for bringing this up.

### PR DESCRIPTION
**Pull request by rauljordan**
Resolves #3949

---

# Description

**Write why you are making the changes in this pull request**

It would be nice to have a way to fetch all blocks during an epoch using a db filter query such as `db.GetBlocks(ctx, filters.NewFilter().SetEpoch(checkpoint.Epoch))`

**Write a summary of the changes you are making**

This PR adds the functionality + tests

### CODE REVIEW

**Comment by rauljordan** [comment]
PTAL @prestonvanloon

File: beacon-chain/db/kv/blocks.go [hunk 1]
```diff
@@ -349,7 +355,13 @@ func (k *Store) SaveGenesisBlockRoot(ctx context.Context, blockRoot [32]byte) er
 // fetchBlockRootsBySlotRange looks into a boltDB bucket and performs a binary search
 // range scan using sorted left-padded byte keys using a start slot and an end slot.
 // If both the start and end slot are the same, and are 0, the function returns nil.
-func fetchBlockRootsBySlotRange(bkt *bolt.Bucket, startSlotEncoded, endSlotEncoded interface{}) [][]byte {
+func fetchBlockRootsBySlotRange(
+	bkt *bolt.Bucket,
+	startSlotEncoded interface{},
```
Comment (prestonvanloon) [inline beacon-chain/db/kv/blocks.go]: Why is it an `interface{}`? Shouldn't it be uint64?

Comment (rauljordan) [reply beacon-chain/db/kv/blocks.go]: It comes from the filters object, which is Enum->interface{}, and then I assert. I wanted to keep that logic within this function as its pretty self-contained

File: beacon-chain/db/kv/blocks_test.go [hunk 2]
```diff
@@ -326,3 +327,35 @@ func TestStore_Blocks_RetrieveRange(t *testing.T) {
 		t.Errorf("Wanted %d, received %d", want, len(retrieved))
 	}
 }
+
+func TestStore_Blocks_Retrieve_Epoch(t *testing.T) {
+	db := setupDB(t)
+	defer teardownDB(t, db)
+	b := make([]*ethpb.BeaconBlock, 500)
```
Comment (shayzluf) [inline beacon-chain/db/kv/blocks_test.go]:
```suggestion
        slots := params.BeaconConfig().SlotsPerEpoch * 7
	b := make([]*ethpb.BeaconBlock, slots)
```

File: beacon-chain/db/kv/blocks_test.go [hunk 3]
```diff
@@ -326,3 +327,35 @@ func TestStore_Blocks_RetrieveRange(t *testing.T) {
...
+	for i := 0; i < 500; i++ {
```
Comment (shayzluf) [inline beacon-chain/db/kv/blocks_test.go]:
```suggestion
	for i := 0; i < slots; i++ {
```

**Pull request merged**